src/components/pin/pin.edgedb.repository.ts

```diff
@@ -0,0 +1,83 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { DateTime } from 'luxon';
+import { ID, PublicOf, Session } from '~/common';
+import { Client } from '~/core/edgedb';
+import { PinRepository } from './pin.repository';
+
+@Injectable()
+export class PinEdgeDBRepository implements PublicOf<PinRepository> {
+  constructor(@Inject(Client) private readonly client: Client) {}
+
+  async isPinned(id: ID, session: Session): Promise<boolean> {
```

Comment:
@CarsonF - Since the schema itself defines a `pinned` property that is automatically inherited by any type that implements the `Pinnable` abstract type, doesn't that make this `isPinned` method unneeded in the EdgeDB world?

Reply:
The `pinned` property is a computed shortcut. 
`User.pins` holds the links for the `Pinnable` objects that user has pinned.
So this could query cast the ID to the `Pinnable` instance and then check the `pinned` property on it.
That uses the current user global which has already been connected in the app code in another layer.
```edgeql
select (<Pinnable><uuid>$id).pinned
```
Or
```edgeql
with obj := <Pinnable><uuid>$id
select obj.pinned
```